app/internal/page/how_should_replacement_attorneys_step_in.go

```diff
@@ -9,16 +9,84 @@ import (
 type howShouldReplacementAttorneysStepInData struct {
 	App    AppData
 	Errors map[string]string
+	Form   *howShouldReplacementAttorneysStepInForm
+}
+
+type howShouldReplacementAttorneysStepInForm struct {
+	WhenToStepIn string
+	OtherDetails string
 }
 
 func HowShouldReplacementAttorneysStepIn(tmpl template.Template, lpaStore LpaStore) Handler {
 	return func(appData AppData, w http.ResponseWriter, r *http.Request) error {
-		// Todo in next ticket
+		lpa, err := lpaStore.Get(r.Context(), appData.SessionID)
+		if err != nil {
+			return err
+		}
 
 		data := &howShouldReplacementAttorneysStepInData{
 			App: appData,
+			Form: &howShouldReplacementAttorneysStepInForm{
+				WhenToStepIn: lpa.HowShouldReplacementAttorneysStepIn,
+				OtherDetails: lpa.HowShouldReplacementAttorneysStepInDetails,
+			},
+		}
+
+		if r.Method == http.MethodPost {
+			data.Form = readHowShouldReplacementAttorneysStepInForm(r)
+			data.Errors = data.Form.Validate()
+
+			if len(data.Errors) == 0 {
+				lpa.HowShouldReplacementAttorneysStepIn = data.Form.WhenToStepIn
+
+				if data.Form.WhenToStepIn != "other" {
+					lpa.HowShouldReplacementAttorneysStepInDetails = ""
+				} else {
+					lpa.HowShouldReplacementAttorneysStepInDetails = data.Form.OtherDetails
+				}
+
+				lpa.HowShouldReplacementAttorneysStepInDetails = data.Form.OtherDetails
```

Comment:
I guess you meant to delete the `if-else` above?

Reply:
The intent here was for people coming back and changing their response to include something other than `other`. I think what needs to happen is dropping line 48 so we only set the details when the response is `other`. This will avoid persisting `OtherDetails` that's prepopulated in the form when they first hit the page.

Reply:
I thought because of the way the validation works you'd only need line 48 since selecting other validates you must have details, selecting non-other validates details is empty.